FlowCrypt/Controllers/Setup/SetupInitialViewController.swift

```diff
@@ -57,25 +59,31 @@ final class SetupInitialViewController: TableNodeViewController {
     private let decorator: SetupViewDecorator
     private let clientConfiguration: ClientConfiguration
     private let emailKeyManagerApi: EmailKeyManagerApiType
+    private let appContext: AppContext
 
     private lazy var logger = Logger.nested(in: Self.self, with: .setup)
 
     init(
+        appContext: AppContext,
         user: UserId,
-        backupService: BackupServiceType = BackupService(),
+        backupService: BackupServiceType? = nil,
         router: GlobalRouterType = GlobalRouter(),
         decorator: SetupViewDecorator = SetupViewDecorator(),
-        clientConfigurationService: ClientConfigurationServiceType = ClientConfigurationService(),
-        emailKeyManagerApi: EmailKeyManagerApiType = EmailKeyManagerApi()
+        emailKeyManagerApi: EmailKeyManagerApiType? = nil
     ) {
+        self.appContext = appContext
         self.user = user
+        let backupService = backupService ?? BackupService(
+            backupProvider: appContext.getRequiredMailProvider().backupProvider,
+            messageSender: appContext.getRequiredMailProvider().messageSender
+        )
```

Comment:
Notice how backupService is initialized with nil first and only then initialized properly. This is an "optional parameter" and it's done this way because the initialization needs another parameter which cannot be used directly above.

I've written this pattern probably 100x in this PR and it bothers me.

It's only ever needed if we wanted to unit-test the VC but we don't really unit test VCs. So with a bit of luck, I may rip all of this out and make most VC parameters required and supplied from context or from whoever calls the VC. That should clean things up.